peps/pep-0735.rst

```diff
@@ -0,0 +1,476 @@
+  the generated lockfiles can be referenced by the names of their originating
+  dependency groups in ``pyproject.toml``.
+* *Input data* to a ``tox``, ``nox``, or ``hatch`` environment, as can
+  currently be achieved, for example, with ``deps = -r requirements.txt`` in
+  ``tox.ini``. These tools will need to add additional options for processing
+  Dependency Groups.
+* Embeddable data for ``pyproject.toml`` within a script (as in PEP 723). This
+  PEP does not define exactly how PEP 723 should be modified, but being
+  consumable by that interface is a stated goal.
+* IDE discovery of requirements data. e.g. VSCode could look for a dependency
+  group named ``test`` to use when running tests.
+
+Regarding Poetry and PDM Dependency Groups
+------------------------------------------
+
+Poetry and PDM already offers a feature which each calls "Dependency Groups",
+but using non-standard data belonging to the ``poetry`` and ``pdm`` tools.
+(PDM also uses extras for some Dependency Groups, and overlaps the notion
+heavily with extras.)
+
+This PEP is not guaranteed to be a perfectly substitutable solution for the
+same problem space for each tool. However, the ideas are extremely similar, and
+it should be possible for Poetry and PDM to support at least some
+PEP-735-standardized Dependency Group configurations using their own Dependency
+Group nomenclature.
+
+A level of interoperability with Poetry and PDM is a goal of this PEP, but
+certain features and behaviors defined here may not be supported by Poetry and
+PDM. Matching the existing Poetry and PDM *semantics* for Dependency Groups is
+a non-goal.
+
+Dependency Groups are not Hidden Extras
+---------------------------------------
+
+One could be forgiven for thinking that Dependency Groups are just extras which
+go unpublished.
+
+However, there are two major features they have which distinguish them from
+extras:
+
+* they support non-package projects
+
+* installation of a dependency group does not imply installation of a package's
+  dependencies (or the package itself)
+
+Specification
+=============
+
+This PEP defines a new section (table) in ``pyproject.toml`` files named
+``requirements``. The ``requirements`` table contains exactly one key,
+``packages``, which is a table. All other keys in ``requirements`` are reserved
+for future use.
+
+The ``packages`` table contains an arbitrary number of user-defined keys, each of
+which has, as its value, a list of requirements specifiers (defined below).
+These keys must match the following regular expression:
+``[a-z0-9][a-z0-9-]*[a-z0-9]``. Meaning that they must be all lower-case
+alphanumerics, with ``-`` allowed only in the middle, and at least
+two characters long. These requirements are chosen so that the normalization
+rules used for pypi package names do not apply, as the names are already
+normalized.
+
+Requirements specifiers will use a definition which extends PEP 508. Therefore,
```

Comment:
It's better to reference the specs at packaging.python.org rather than individual PEPs.

Reply:
I'll do a pass for this. I've just incorporated the stylistic fix feedback to use the `:pep:` sphinx role, which I didn't know about. But I'd like to move any PEP numbers into footnote references, and refer to spec docs if that's the preferred point of reference.

Reply:
Generally, we prefer inline links rather than footnotes (they save a click), unless you have something extra to say in the footnote.

Perhaps something like "the X specification (PEP XXX)"?

Reply:
I've done a pass to try to ensure that PEP numbers never appear without a descriptive name, so I'm marking this resolved, but I wouldn't be surprised if this can be refined further.